lib/rails/redirect_with_flash.rb

```diff
@@ -0,0 +1,43 @@
+Synvert::Rewriter.new 'rails', 'redirect_with_notice' do
+  description <<-EOS
+
+  Fold flash setting into redirect_to.
+
+  flash[:notice] = "huzzah"
+  redirect_to root_path
+  =>
+  redirect_to root_path, notice: "huzzah"
+
+  and
+
+  flash[:error] = "booo"
+  redirect_to root_path
+  =>
+  redirect_to root_path, flash: {error: "huzzah"}
+
+  EOS
+  within_file 'app/controllers/**/*rb' do
+    within_node type: 'def' do
+      line = nil
+      msg = nil
+      remover_action = nil
+      flash_type = nil
+      with_node type: 'send', receiver: 'flash', arguments: {size: 2, first: %r{:notice|:error}, last: {type: :str}} do
```

Comment:
why do you just check `notice` and `error`? it could be `alert` or anything else